gemrb/plugins/SDLAudio/SDLAudio.cpp

```diff
@@ -73,7 +73,28 @@ bool SDLAudio::Init(void)
 	return true;
 }
 
-void SDLAudio::music_callback(void *udata, unsigned short *stream, int len) {
+void SDLAudio::SetAudioStreamVolume(uint8_t *stream, int len, int volume)
+{
+	uint8_t *mixData = new uint8_t[len];
+	memcpy(mixData, stream, len * sizeof(uint8_t));
+	memset(stream, 0, len); // mix audio data against silence
+	SDL_MixAudio(static_cast<uint8_t*>(stream), static_cast<const uint8_t*>(mixData), len, volume);
+	delete[] mixData;
+}
+
+void SDLAudio::music_callback(void *udata, uint8_t *stream, int len)
+{
+	ieDword volume = 100;
+	core->GetDictionary()->Lookup("Volume Music", volume);
+
+	// No point of bothering if it's off anyway. Hope nothing breaks..
+	if (volume == 0) {
+		return;
```

Comment:
isn't this just going to leave the stream filled with whatever was in it before? I guess it doesn't matter?

Reply:
SDL_mixer 2 resets it to internally 0 before music callback.
SDL_mixer 1.2 doesn't, but it seems to be working fine, so my guess is that it's taken care of somewhere in SDL too.

Reply:
cool